test/functional/test_runner.py

```diff
@@ -471,6 +474,13 @@ def main():
 
     logging.debug("Temporary test directory at %s" % tmpdir)
 
+    results_filepath = None
+    if args.resultsfile:
+        results_filepath = pathlib.Path(args.resultsfile)
+        # Stop early if the parent directory doesn't exist
+        assert results_filepath.parent.exists()
```

Comment:
Would be useful to have a log come from this assert
```suggestion
        assert results_filepath.parent.exists(), "Results file path parent directory does not exist"
```
Before:
```
test/functional/test_runner.py --resultsfile ./fakedir/functional_test_results.csv feature_blocksdir
Temporary test directory at /tmp/test_runner_₿_🏃_20240616_141849
Traceback (most recent call last):
  File "/home/kevkevin/DEVDIR/bitcoin/test/functional/test_runner.py", line 939, in <module>
    main()
  File "/home/kevkevin/DEVDIR/bitcoin/test/functional/test_runner.py", line 481, in main
    assert results_filepath.parent.exists()
AssertionError
```
After:
```
test/functional/test_runner.py --resultsfile ./fakedir/functional_test_results.csv feature_blocksdir
Temporary test directory at /tmp/test_runner_₿_🏃_20240616_142329
Traceback (most recent call last):
  File "/home/kevkevin/DEVDIR/bitcoin/test/functional/test_runner.py", line 939, in <module>
    main()
  File "/home/kevkevin/DEVDIR/bitcoin/test/functional/test_runner.py", line 481, in main
    assert results_filepath.parent.exists(), "Results file path parent directory does not exist"
AssertionError: Results file path parent directory does not exist
```

Reply:
Thanks.  This is an improvement.  Included.